sympy/core/add.py

```diff
@@ -1011,11 +1011,18 @@ def _eval_as_leading_term(self, x, cdir=0):
         if not new_expr:
             # simple leading term analysis gave us cancelled terms but we have to send
             # back a term, so compute the leading term (via series)
+            from sympy.functions import Piecewise, piecewise_fold
             n0 = min.getn()
             res = Order(1)
             incr = S.One
+
+            if old.has(Piecewise):
+                expr = piecewise_fold(old)
```

Comment:
Is there an example where this is needed?

Reply:
It is directly needed in the example in 9449 - 
`-Abs(_w - eps) + Abs(_w + eps)` changes to 
`-Piecewise((_w, Eq(eps, 0)), (-_w*sign(eps) + eps*sign(eps), True)) + Piecewise((_w, Eq(eps, 0)), (_w*sign(eps) + eps*sign(eps), True))`,
and since this value is not `piecewise_fold`ed, there is in infinite recursion in the while loop just below

Reply:
I think that the leading term method of Piecewise should handle this.

Reply:
What we want is for 
```
-Piecewise((_w, Eq(eps, 0)), (-_w*sign(eps) + eps*sign(eps), True)) + Piecewise((_w, Eq(eps, 0)), (_w*sign(eps) + eps*sign(eps), True))
```
to be folded into
```
Piecewise((0, Eq(eps, 0)), (2*_w*sign(eps), True))
```
For this, I think it is necessary for the caller to `piecewise_fold` the expression, as the individual `Piecewise` terms cannot be folded but the addition of both can.

Reply:
OK. The sum has to be simplified.